torch_harmonics/resample.py

```diff
@@ -67,48 +67,51 @@ def __init__(
 
         # for upscaling the latitudes we will use interpolation
         self.lats_in, _ = _precompute_latitudes(nlat_in, grid=grid_in)
-        self.lons_in = np.linspace(0, 2 * math.pi, nlon_in, endpoint=False)
+        self.lons_in = torch.linspace(0, 2 * math.pi, nlon_in+1)[:-1]
         self.lats_out, _ = _precompute_latitudes(nlat_out, grid=grid_out)
-        self.lons_out = np.linspace(0, 2 * math.pi, nlon_out, endpoint=False)
+        self.lons_out = torch.linspace(0, 2 * math.pi, nlon_out+1)[:-1]
 
         # in the case where some points lie outside of the range spanned by lats_in,
         # we need to expand the solution to the poles before interpolating
         self.expand_poles = (self.lats_out > self.lats_in[-1]).any() or (self.lats_out < self.lats_in[0]).any()
         if self.expand_poles:
-            self.lats_in = np.insert(self.lats_in, 0, 0.0)
-            self.lats_in = np.append(self.lats_in, np.pi)
+            self.lats_in = torch.cat([torch.tensor([0.], dtype=torch.float64),
+                                      self.lats_in,
+                                      torch.tensor([math.pi], dtype=torch.float64)]).contiguous()
+            #self.lats_in = np.insert(self.lats_in, 0, 0.0)
+            #self.lats_in = np.append(self.lats_in, np.pi)
 
         # prepare the interpolation by computing indices to the left and right of each output latitude
-        lat_idx = np.searchsorted(self.lats_in, self.lats_out, side="right") - 1
+        lat_idx = torch.searchsorted(self.lats_in, self.lats_out, side="right") - 1
         # make sure that we properly treat the last point if they coincide with the pole
-        lat_idx = np.where(self.lats_out == self.lats_in[-1], lat_idx - 1, lat_idx)
+        lat_idx = torch.where(self.lats_out == self.lats_in[-1], lat_idx - 1, lat_idx)
 
         # lat_idx = np.where(self.lats_out > self.lats_in[-1], lat_idx - 1, lat_idx)
         # lat_idx = np.where(self.lats_out < self.lats_in[0], 0, lat_idx)
 
         # compute the interpolation weights along the latitude
-        lat_weights = torch.from_numpy((self.lats_out - self.lats_in[lat_idx]) / np.diff(self.lats_in)[lat_idx]).float()
+        lat_weights = ((self.lats_out - self.lats_in[lat_idx]) / torch.diff(self.lats_in)[lat_idx]).to(torch.float32)
         lat_weights = lat_weights.unsqueeze(-1)
 
         # convert to tensor
-        lat_idx = torch.LongTensor(lat_idx)
+        #lat_idx = torch.LongTensor(lat_idx)
```

Comment:
is this unnecessary here? If so we should delete it. Same for distributed case.

Reply:
I think it is already a long tensor

Reply:
done